app/services/notifications/fire_service.rb

```diff
@@ -0,0 +1,45 @@
+module Notifications
+  class FireService
+    def initialize(notification)
+      @notification = notification
+    end
+
+    def fire
+      return if @notification.blank?
+
+      return if @notification.recipient.webpush_subscription.blank?
+
+      begin
+        Webpush.payload_send(
+          message: JSON.generate(message),
+          endpoint: @notification.recipient.webpush_subscription['endpoint'],
+          p256dh: @notification.recipient.webpush_subscription['p256dh'],
+          auth: @notification.recipient.webpush_subscription['auth'],
+          vapid: vapid_keys,
+          ssl_timeout: 5,
+          open_timeout: 5,
+          read_timeout: 5
+        )
+      rescue
```

Comment:
A blanket `rescue` like this is pretty-much always a red-flag. _Any_ failure, including a temporary network failure might cause this user's subscription to be deleted, right? 🤔